plugins/com.jboss.devstudio.core.cheatsheet/src/com/jboss/devstudio/core/cheatsheet/CheatsheetStartup.java

```diff
@@ -0,0 +1,91 @@
+ ************************************************************************************/
+package com.jboss.devstudio.core.cheatsheet;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.osgi.service.datalocation.Location;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.cheatsheets.OpenCheatSheetAction;
+import org.osgi.service.prefs.BackingStoreException;
+
+public class CheatsheetStartup implements org.eclipse.ui.IStartup {
+	
+	public static final String PLUGIN_ID = "com.jboss.devstudio.core.cheatsheet";
+	private static final String RELATIVE_ROOT_PATH = (Platform.getOS().equals(Platform.OS_MACOSX) ? "../../../" //$NON-NLS-1$
+			: ""); //$NON-NLS-1$
+	private static final String CHEATSEET_PATH = RELATIVE_ROOT_PATH
+			+ "../../cheatsheets/guided-development.xml"; //$NON-NLS-1$
+
+	@Override
+	public void earlyStartup() {
+		File pathToXml;
+		try {
+			pathToXml = CheatsheetStartup.getRelativeDirectory(CheatsheetStartup.CHEATSEET_PATH);
+			if (CheatsheetStartup.isFirstStart() && pathToXml.canRead()) {
+				Debug.trace("File '" + pathToXml + "' exists, running async exec to open it.");
+				Display.getDefault().asyncExec(() -> {
+					try {
+						CheatsheetStartup.setFirstStart(false);
+						new OpenCheatSheetAction("guided.development", "Guided Development", pathToXml.toURI().toURL())
+								.run();
+						Debug.trace("File '" + pathToXml + "' opened.");
+					} catch (MalformedURLException e) {
+						logError(e.getMessage(), e);
+					}
+				});
+			} else {
+				Debug.trace("Skip opening of guided development cheatsheet");
+			}
+		} catch (IOException ex) {
+			logError("Configuration folder location cannot be resolved", ex);
+		}
+	}
+
+	private static String getConfiguration() throws IOException {
+		Location configLocation = Platform.getConfigurationLocation();
+		URL configURL = configLocation.getURL();
+		return FileLocator.resolve(configURL).getPath();
+	}
+
+	private static File getRelativeDirectory(String dir) throws IOException {
+		String configuration = getConfiguration();
+		return new File(configuration, dir);
+	}
+	
+	public static void logError(String message, Throwable exception) {
+		Platform.getLog(Platform.getBundle(PLUGIN_ID)).log(new Status(Status.ERROR,
+				PLUGIN_ID, message, exception));
+	}
+
+	public static boolean isFirstStart() {
+		IEclipsePreferences prefs = InstanceScope.INSTANCE.getNode(CheatsheetStartup.PLUGIN_ID);
+		return prefs.getBoolean("FIRST_START", true);
+	}
+
+	public static void setFirstStart(boolean value) {
+		IEclipsePreferences prefs = InstanceScope.INSTANCE.getNode(CheatsheetStartup.PLUGIN_ID);
```

Comment:
Cheetsheet is opened for every new workspace. Can you please add option to disable/reenable opening the cheetsheet ? Cheetsheet could contain a checkbox "Do not open for new workspace" or/and implement Preference Page where user can disable/enable the opening.

Reply:
I can use configuration scope instead for this release, that means it will be open only once and only in first workspace. WDYT?

Reply:
If I have to choose between Configuration and instance scope, i would choose to keep it how it is now (Instance) and implement the Preference Page later.

But the next release is in 3 months.. @dgolovin if you do not have time to implement the Preference Page before the release I can help

Reply:
> Cheetsheet could contain a checkbox "Do not open for new workspace" or/and implement Preference Page where user can disable/enable the opening.

@rawagner AFIK there is no way to do it in Cheatsheet view. To suppress opening on new workspace would still require Configuration scope preference to check that Guided Development cheatsheet should not be opened when new workspace created.

> But the next release is in 3 months.. @dgolovin if you do not have time to implement the Preference Page before the release I can help

Thanks for willing to help, but I would rather spend more time on creating cheatsheets for CDK rather than blowing up code related to eclipse preferences. Because creating new preference page would require decisions on how to name it and where to put it in Preference pages hierarchy (there is no devstudio category in preferences), documentation update for all of it.